### PR TITLE
transform: allow transforms to hold thread local state

### PR DESCRIPTION
**Pull request by petrosagg**
### Motivation

The bounds seem to be been added for constructing a lazy static vector but the same can be accomplished with the thread_local macro

### CODE REVIEW

File: src/transform/tests/test_runner.rs [hunk 1]
```diff
@@ -137,8 +136,8 @@ mod tests {
         let format_type = get_format_type(args);
 
         let out = match test_type {
-            TestType::Opt => {
-                for transform in FULL_TRANSFORM_LIST.iter() {
+            TestType::Opt => FULL_TRANSFORM_LIST.with(|transforms| -> Result<_, Error> {
```
Comment (wangandi) [inline src/transform/tests/test_runner.rs:139]: As a side note, where do I find the documentation for this `with` method?

**Review by wangandi — APPROVED** [review]
Thanks for making the test be less invasive!

Looks like you need to add Cargo.lock to this PR to have it pass tests.

File: src/transform/tests/test_runner.rs (hunk 1 above)
Comment (petrosagg) [reply src/transform/tests/test_runner.rs:139]: Here https://doc.rust-lang.org/std/thread/struct.LocalKey.html#method.with

**Comment by petrosagg** [comment]
@wangandi thanks for the quick review!

**Pull request merged**